webui: Allow grace login limit

carma12 opened the pull request:
There was no support for setting the grace login limit on the WebUI. The
only way to so was only via CLI:

   `ipa pwpolicy-mod --gracelimit=2 global_policy`

Thus, the grace login limit must be updated from the policy section and
this will reflect also on the user settings (under the 'Password Policy'
section)

Fixes: https://pagure.io/freeipa/issue/9211

Signed-off-by: Carla Martinez <carlmart@redhat.com>

rcritten commented on install/ui/src/freeipa/policy.js:
```diff
@@ -72,6 +72,10 @@ return {
+                        },
+                        {
+                            name: 'passwordgracelimit',
+                            measurement_unit: 'number_of_passwords'
```
The unit isn't quite right. It defines the number of LDAP BIND that are allowed post-expiration so that the password can be reset by the user. I'm struggling with a pity wording for it though :-) maybe "grace logins" is sufficient.

netoarmando replied:
After deciding on the wording that must be added to the i18n messages dict:

https://github.com/freeipa/freeipa/blob/c2e79fa0b64be5d51d3ef0e70bb8671cbe803657/ipaserver/plugins/internal.py#L417-L420

And i18n object used for testing:

https://github.com/freeipa/freeipa/blob/c2e79fa0b64be5d51d3ef0e70bb8671cbe803657/install/ui/test/data/i18n_messages.json#L214-L217

carma12 replied:
"Grace logins" sounds good to me, I will do the changes to the files mentioned above. If we come up with a better name, I can change it later :)

flo-renaud commented:
Hi @carma12 

The field for grace login limit displays empty parentheses because there is no unit set:
![image](https://user-images.githubusercontent.com/21310191/182831318-9923325c-9d03-4b6d-b768-d1acecafb692.png)

Is it possible to remove these empty parentheses?

Otherwise, the WebUI behaves properly:
- [x] from the webui change grace login limit, properly updates the LDAP entry
- [x] change the grace limit using the CLI, refresh the webui page, properly displays the new value
This is working for both global_policy and custom group policies.

carma12 commented:
Hi @flo-renaud 

I had a typo in my code when defining the measurement units on the user's settings and the policy page. I fixed that and now the correct measurement unit (`grace logins`) is displayed in the parenthesis.
![image](https://user-images.githubusercontent.com/8112750/182872052-962f6702-7028-473c-927c-37a3c65baed2.png)

After the fix, the parenthesis is not empty anymore so I left it as is. Should we keep this measurement unit?

flo-renaud commented:
Hi @carma12 
> Hi @flo-renaud
> 
> I had a typo in my code when defining the measurement units on the user's settings and the policy page. I fixed that and now the correct measurement unit (`grace logins`) is displayed in the parenthesis. ![image](https://user-images.githubusercontent.com/8112750/182872052-962f6702-7028-473c-927c-37a3c65baed2.png)
> 
> After the fix, the parenthesis is not empty anymore so I left it as is. Should we keep this measurement unit?

Is it possible to simply display (no parenthesis, no unit): Grace login limit
the same way as we do for instance for Max failures:

![image](https://user-images.githubusercontent.com/21310191/182874676-3154e242-2650-4b84-bccf-14115d45e649.png)

carma12 commented:
Hi @flo-renaud 

I updated my code and removed the measurement unit from the policy and the user settings page. Now only the name is displayed:
![image](https://user-images.githubusercontent.com/8112750/182883605-60d83ce2-fb6e-44d2-b102-23585524d576.png)

flo-renaud commented:
Hi @carma12 
thanks for the update, works for me. ACK.
Special kudos as it's your first commit in the project and you followed the commit message rules :)

carma12 commented:
master:

* 42b2607a6f3809ba26368f415c04aefa31728137 webui: Allow grace login limit
